### ISSUE TITLE
opsz dogma/rationale

### ISSUE THREAD

**Issue by laerm0** [googlefonts/fontbakery#2708]
The text for the `opsz` coordinate check is confusing, largely because the source text (Microsoft OT spec variable axes) is confusing:

[FB docs](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/fvar.html#com.google.fonts/check/varfont/regular_opsz_coord):
> The variable font ‘opsz’ (Optical Size) axis coordinate should be between 9 and 13 on the ‘Regular’ instance.

Rationale from Microsoft:
>According to the Open-Type spec’s registered design-variation tag ‘opsz’ available at https://docs.microsoft.com/en-gb/typography/opentype/spec/dvaraxistag_opsz
>If a variable font has a ‘opsz’ (Optical Size) axis, then the coordinate of its ‘Regular’ instance is recommended to be a value in the range 9 to 13.

"Regular" as an instance name on the optical size axis doesn't make any sense. "Regular" is the 400 weight and has nothing to do with the optical size. This implies that a 400 weight on the 9-13 opsz axis would be called "`$fontname` Regular Regular." 

I know we've had this discussion before, but I am still very unsatisfied. I can't recall a single example of a font that was released before variable font tech was created where the smallest size is called the regular: it's called something like "Text" or even just by the intended point size. I believe there was some discussion that naming optical size weights by point size is less logical in a digital type world, but that doesn't stop us from using the traditional "Text"/"Display" model of optical size naming. The spec does not say that the "regular" instance should be _named_ "Regular"; my interpretation is that the origin should be in the 9 to 13 range.

If we're just going to say in the FB docs that the GF spec is this because Microsoft/developers of the OT spec say so, then that's a thing and so it is, but I really think we ought to acknowledge the inherent confusion in such a naming scheme somewhere. At the very least, a font without an `opsz` instance named Regular should be given a `WARN` and not a `FAIL`.

**Comment by felipesanches** [comment]
What would you suggest here as a solution to this?

**Comment by dberlow** [comment]
I don’t think that the instance along the optical size axis that is the default weight, width or any other axes, has a default size of “regular”. I believe it is regular and the optical size value number. As for what that number is, I have defined 14 in two ongoing vfs, with no complaints from clients,  authors of the spec, or applications.

**Comment by laerm0** [comment]
I'm not going to call it _a solution_… My advice would be to explain that fontbakery's check is adhering to the OT spec and any confusion is due to that, but that GF has a "recommended interpretation" of that part of the spec which could be something like:

>The origin of your optical size axis should be between 9-13. This number can be interpreted as a suggested point size for this instance's usage. This has been called the "Text" version of a typeface throughout history. Similarly, the higher end of the optical size axis is often called "Display" and is traditionally designed for a point size around 72.

And I'd suggest that the tolerance range for the "regular" instance of an `opsz` axis be a little wider than 9-13: there are fonts designed for 7pt or lower, so a range of 7-14 seems reasonable for the origin of an `opsz` axis. Maybe make a `WARN` be an `opsz` origin below 8 and in the 15 – 20 range and call it a `FAIL` if the `opsz` origin is >20 and/or the scale is absurd (i.e. not in a range below 100, as if the designer used the range of a `wght` axis for `opsz`).

@dberlow is pretty much spot-on, I think (and please correct me on my interpretation of sizes and nomenclature if you feel it's necessary, David).

**Comment by laerm0** [comment]
This spec is getting more confusing: the text makes no sense.

```
    * FAIL: Instance name "ExtraLight Text" is incorrect. It should be "ExtraLight" [code: bad-name]
    * WARN: VF instance string 'ExtraLight Text' contains the word 'Text' or 'Display'. We prefer optical sizes to use pt sizes e.g '16pt ExtraLight'.
```

Okay, so it's a `FAIL` because of ???, but the `WARN` seems to explain why, but then says "we prefer": a preference is not a `FAIL`. 

Totally not on you, Felipe. Can I get a word from @davelab6 ?

**Comment by m4rc1e** [comment]
Why not do a git blame on the check? you'll see I'm the culprit. You're getting a fail because the name parser isn't tokenising "Text". I'll add this soon.

For opsz, we don't name them Text or Display. Naming was discussed in this issue, https://github.com/googlefonts/fontbakery/issues/2608#issuecomment-522435701

**Comment by m4rc1e** [comment]
> You're getting a fail because the name parser isn't tokenising "Text". I'll add this soon.

Spoke too soon, it's already implemented.

https://github.com/googlefonts/fontbakery/blob/master/Lib/fontbakery/profiles/googlefonts.py#L4045-L4048

Please update your fontbakery

**Comment by laerm0** [comment]
1. I didn't know about `git blame`
2. My immediate issue was with the wording not making sense, that `FAIL` shouldn't seem to be because of a preference
3. My greater issue is with, yes, the entirety of this naming scheme

**Comment by laerm0** [comment]
…so I'll close this in favor of taking my discussion to #2608

**Comment by davelab6** [comment]
GitHub has a UI for blame too, it's nice :)

I believe the FAIL is because only weights are style strings, everything else has to be in the family string

**Comment by m4rc1e** [comment]
I see now, it's raising both a FAIL and a WARN if an instance name contains either "Display" or "Text". Dave, since we want opsz to use point sizes, shall this check raise a FAIL only?
